key/keybook.go

```diff
@@ -0,0 +1,175 @@
+package key
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+
+	pstore "github.com/libp2p/go-libp2p-core/peerstore"
+)
+
+type memoryKeyBook struct {
+	sync.RWMutex // same lock. wont happen a ton.
+	pks          map[peer.ID]ic.PubKey
+	sks          map[peer.ID]ic.PrivKey
+}
+
+var _ pstore.KeyBook = (*memoryKeyBook)(nil)
+
+// noop new, but in the future we may want to do some init work.
```

Comment:
allocation of the maps is initialization work

Reply:
yeah, this was simply ripped out from the peerstoremem package. I have to work on this piece a bit to get it in line with our code.